Tests/RoslynPluginGeneratorTests/NuGetPackageHandlerTests.cs

```diff
@@ -17,22 +19,211 @@ public class NuGetPackageHandlerTests
+            };
+            buildPackage(dependentMetadata, dependentPackageFile);
+
+            TestLogger logger = new TestLogger();
+            NuGetPackageHandler handler = new NuGetPackageHandler(logger);
+
+            // Act
+            // Attempt to download a package which is not released with a dependency that is released
+            IPackage package = handler.FetchPackage(testDir, dependentPackageName, null, testDownloadDir);
+
+            // Assert
+            // Package should have been downloaded
+            AssertPackageDownloaded(testDownloadDir, dependentPackageName);
+            Assert.IsNotNull(package, "Expected a reference to the package to be returned");
+        }
+
+        [TestMethod]
+        public void NuGet_TestPackageDownload_PreRelease_PreRelease()
+        {
+            // Arrange
+            string testDir = TestUtils.CreateTestDirectory(this.TestContext);
+            string testDownloadDir = Path.Combine(testDir, "download");
+
+            // Create test NuGet payload and packages
+            string testPackageFile = Path.Combine(testDir, "testPackage.nupkg");
+            string dependentPackageFile = Path.Combine(testDir, "dependentPackage.nupkg");
+
+            ManifestMetadata testMetadata = new ManifestMetadata()
+            {
+                Authors = "Microsoft",
+                Version = preReleaseVersion,
+                Id = testPackageName,
+                Description = "A description",
+            };
+            buildPackage(testMetadata, testPackageFile);
+
+            List<ManifestDependencySet> dependencies = new List<ManifestDependencySet>()
+            {
+                new ManifestDependencySet()
+                {
+                    Dependencies = new List<ManifestDependency>()
+                    {
+                        new ManifestDependency()
+                        {
+                            Id = testPackageName,
+                            Version = preReleaseVersion,
+                        }
+                    }
+                }
+            };
+            ManifestMetadata dependentMetadata = new ManifestMetadata()
+            {
+                Authors = "Microsoft",
+                Version = preReleaseVersion,
+                Id = dependentPackageName,
+                Description = "A description",
+                DependencySets = dependencies,
+            };
+            buildPackage(dependentMetadata, dependentPackageFile);
+
+            TestLogger logger = new TestLogger();
+            NuGetPackageHandler handler = new NuGetPackageHandler(logger);
+
+            // Act
+            // Attempt to download a package which is not released with a dependency that is not released
+            IPackage package = handler.FetchPackage(testDir, dependentPackageName, null, testDownloadDir);
+
+            // Assert
+            // Package should have been downloaded
+            AssertPackageDownloaded(testDownloadDir, dependentPackageName);
```

Comment:
Is there any state where the package doesn't get downloaded, e.g. a package with a dependency that's not in NuGet or when the package itself is not present. Is this a good place for these kinds of tests or have these scenarios been tested elsewhere?